datachannel.go

```diff
@@ -24,6 +24,27 @@ type Config struct {
 	Label                string
 }
 
+func newDataChannel(stream *sctp.Stream, config *Config) *DataChannel {
+	switch config.ChannelType {
+	case ChannelTypeReliable:
+		stream.SetReliabilityParams(false, sctp.ReliabilityTypeReliable, config.ReliabilityParameter)
+	case ChannelTypeReliableUnordered:
+		stream.SetReliabilityParams(true, sctp.ReliabilityTypeReliable, config.ReliabilityParameter)
+	case ChannelTypePartialReliableRexmit:
+		stream.SetReliabilityParams(false, sctp.ReliabilityTypeRexmit, config.ReliabilityParameter)
+	case ChannelTypePartialReliableRexmitUnordered:
+		stream.SetReliabilityParams(true, sctp.ReliabilityTypeRexmit, config.ReliabilityParameter)
+	case ChannelTypePartialReliableTimed:
+		stream.SetReliabilityParams(false, sctp.ReliabilityTypeTimed, config.ReliabilityParameter)
+	case ChannelTypePartialReliableTimedUnordered:
+		stream.SetReliabilityParams(true, sctp.ReliabilityTypeTimed, config.ReliabilityParameter)
+	default:
```

Comment:
This default case should throw an error (in case we get a config.ChannelType that is in a bad state)